src/FilterByDevice.java

```diff
@@ -0,0 +1,17 @@
+import products.Product;
+
+import java.util.Objects;
+
+public class FilterByDevice implements ProductFilter {
+
+    private final String input;
```

Comment:
non credo ci sia bisogno di dichiararla come variabile d'istanza in quanto se la prende già come argomento

Reply:
se non la dichiari non puoi inserirla nel costruttore col this.input = input.toLowerCase.